pyhealth/unittests/test_datasets/test_mimic4

```diff
@@ -0,0 +1,96 @@
+import datetime
+import unittest
+
+from pyhealth.datasets import MIMIC4Dataset
+
+# this test suite verifies the MIMIC4 demo dataset is parsed correctly and produces
+# the correct dataset for demoing. To qualify the units under test we check the dataset statistics, 
+# and a single sample from the dataset.
+# 
+# the dataset can be found at https://physionet.org/content/mimic-iv-demo/2.2/
```

Comment:
We should put this behind a google api to make things trivial. let me know if there is some way I can help with that.